crates/ol/stf/src/assembly.rs

```diff
@@ -0,0 +1,266 @@
+//! Block assembly flows.
+// TODO should this be in another crate?
+
+use strata_asm_common::AsmManifest;
+use strata_identifiers::Buf32;
+use strata_ledger_types::StateAccessor;
+use strata_merkle::{BinaryMerkleTree, Sha256Hasher};
+use strata_ol_chain_types_new::{
+    OLBlockBody, OLBlockHeader, OLL1ManifestContainer, OLL1Update, OLLog, OLTransaction,
+    OLTxSegment, TransactionAttachment, TransactionPayload,
+};
+
+use crate::{
+    chain_processing,
+    context::{BasicExecContext, BlockContext, TxExecContext},
+    errors::ExecResult,
+    manifest_processing,
+    output::{ExecOutputBuffer, OutputCtx},
+    transaction_processing,
+    verification::{BlockExecInput, BlockPostStateCommitments},
+};
+
+/// Block execution outputs.
+///
+/// These can be used to construct a final block.
+#[derive(Clone, Debug)]
+pub struct BlockExecOutputs {
+    post_state_roots: BlockPostStateCommitments,
+    logs: Vec<OLLog>,
+}
+
+impl BlockExecOutputs {
+    fn new(post_state_roots: BlockPostStateCommitments, logs: Vec<OLLog>) -> Self {
+        Self {
+            post_state_roots,
+            logs,
+        }
+    }
+
+    pub fn post_state_roots(&self) -> &BlockPostStateCommitments {
+        &self.post_state_roots
+    }
+
+    pub fn header_post_state_root(&self) -> &Buf32 {
+        self.post_state_roots.header_state_root()
+    }
+
+    pub fn logs(&self) -> &[OLLog] {
+        &self.logs
+    }
+
+    /// Computes the block's logs root from the log.
+    pub fn compute_block_logs_root(&self) -> Buf32 {
+        // This is just a simple binary merkle tree.
+        if self.logs.is_empty() {
+            // Empty tree has null root
+            return Buf32::zero();
```

Comment:
FWIW, ethereum, _I think_, does not return zero in this case. It make be hash(0) or perhaps some minimal size root of all-0 tree. We can leave it be, but if the code below is generic enough, then we may want to discard this special case.